src/cli.js

```diff
@@ -1260,7 +1260,7 @@ export class CLI {
 								appc = (await import(nodeAppc)).default;
 							}
 							if (typeof mod.init === 'function') {
-								mod.init(this.logger, this.config, this, appc);
+								await mod.init(this.logger, this.config, this, appc);
```

Comment:
Is there any hook method other than this that would benefit from being await-d?

Reply:
That should be it. The only things that the CLI loads dynamically are commands and hooks.

Commands have `config()`, `validate()`, and `run()`. CLI v7 supports each of these functions returning a promise. `config()` and `validate()` do not accept a callback. I ran into a problem where I needed `config()` to run detection which is async and so to solve it, `config()` returns a function that is called async to run detection. Bad design. For `run()`, the callback will resolve the promise before the returned promise, but everything works as expected.

Hooks have `init()`. This was the only high-level API that didn't support async, hence this PR.

I don't think it's necessary to make callbacks like option validators async.